src/xaa.ts

```diff
@@ -449,18 +458,19 @@ export async function filter(array: any[], func: Function) {
  * @param valOrFunc value, or callback to get value, to return if `func` throws
  * @returns awaited result, `valOrFunc`, or `await valOrFunc(err)`.
  */
-export async function tryCatch(func: Function, valOrFunc: any): Promise<any> {
-  try {
-    const r = func();
-    return await r;
-  } catch (err) {
-    if (typeof valOrFunc === "function") {
-      const r = valOrFunc(err);
+export async function tryCatch<T, TAlt>(func: () => T,
+  valOrFunc: ValueOrErrorHandler<TAlt>): Promise<T | TAlt> {
+    try {
+      const r = func();
       return await r;
-    }
+    } catch (err) {
+      if (typeof valOrFunc === "function") {
```

Comment:
Change this to `valOfFunc instanceof Function` to remove the need for the `as Function` cast. Also, you might want to update `ValueOrErrorHandler` to:

```TypeScript
type ValueOrErrorHandler<T> = T | Promise<T> | ((err?: Error) => T | Promise<T>);
```